src/lib.rs

```diff
@@ -675,6 +681,12 @@ fn parse_token<'a>(bytes: &mut Bytes<'a>) -> Result<&'a str> {
 
 #[inline]
 fn parse_uri<'a>(bytes: &mut Bytes<'a>) -> Result<&'a str> {
+    let b = next!(bytes);
+    if !is_uri_token(b) {
+        // First char must be a URI char, it can't be a space which would indicate an empty path.
+        return Err(Error::Token);
+    }
+
     simd::match_uri_vectored(bytes);
```

Comment:
I'm unsure if there are performance implications for eating a char before calling the simd routine, or if there's a better way to address this.